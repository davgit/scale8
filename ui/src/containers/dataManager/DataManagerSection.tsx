import { FC } from 'react';
import { useQuery } from '@apollo/client';
import {
    BreadcrumbButtonProps,
    buildDataManagerButtonProps,
} from '../../utils/BreadcrumbButtonsUtils';
import NavDataManagerQuery from '../../gql/queries/NavDataManagerQuery';
import { NavDataManager } from '../../gql/generated/NavDataManager';
import { Section, SectionItem, SectionProps } from '../abstractions/Section';
import { SectionKey } from '../SectionsDetails';
import { useConfigState, useLoggedInState } from '../../context/AppContext';
import { NextRouter, useRouter } from 'next/router';
import { ChildrenAndIdProps } from '../../types/props/ChildrenAndIdProps';
import { CurrentOrgPermissions } from '../../context/OrgUserReducer';
import { buildOrgButtons } from '../global/OrgSection';

export const buildDataManagerButtons = (
    orgs: SectionItem[],
    currentOrg: SectionItem,
    tagManagerId: string,
    dataManagerId: string,
    router: NextRouter,
    orgPermissions: CurrentOrgPermissions,
    useSignup: boolean,
    isCurrentPage = false,
): BreadcrumbButtonProps[] => [
    ...buildOrgButtons(orgs, currentOrg, router, orgPermissions, useSignup, 'Services'),
    buildDataManagerButtonProps(router, dataManagerId, tagManagerId, isCurrentPage),
];

const DataManagerSection: FC<ChildrenAndIdProps> = (props: ChildrenAndIdProps) => {
    const router = useRouter();
    const { id, children } = props;

    const { orgUserState } = useLoggedInState();
    const { useSignup } = useConfigState();

    const sectionProps: SectionProps<NavDataManager> = {
        children,
        sectionKey: SectionKey.dataManager,
        queryResult: useQuery<NavDataManager>(NavDataManagerQuery, {
            variables: { id },
        }),
        buildButtonsProps: (data, orgPermissions) => {
            return buildDataManagerButtons(
                data.me.orgs,
                data.getDataManagerAccount.org,
                data.getDataManagerAccount.org.tag_manager_account?.id ?? '',
                id,
                router,
                orgPermissions,
                useSignup,
                true,
            );
        },
        buildMenuItemsProps: () => [],
        extractOrgUserDetails: (data) => data.getDataManagerAccount.org,
        accountExpireIn: orgUserState?.dataManagerAccount?.trialExpiration ?? undefined,
        accountIsTrial: orgUserState?.dataManagerAccount?.isTrial ?? undefined,
    };

    return <Section<NavDataManager> {...sectionProps} />;
};

export default DataManagerSection;
