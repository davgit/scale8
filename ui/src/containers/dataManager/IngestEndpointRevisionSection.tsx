import { FC } from 'react';
import { useQuery } from '@apollo/client';
import NavIngestEndpointRevisionQuery from '../../gql/queries/NavIngestEndpointRevisionQuery';
import { NavIngestEndpointRevision } from '../../gql/generated/NavIngestEndpointRevision';
import { buildIngestEndpointRevisionButtonProps } from '../../utils/BreadcrumbButtonsUtils';
import { Section, SectionProps } from '../abstractions/Section';
import { buildDuplicateAction } from '../../utils/TableActionsUtils';
import { pageActions } from '../../actions/PageActions';
import { extractPermissionsFromOrgUser } from '../../context/OrgUserReducer';
import { SectionKey } from '../SectionsDetails';
import { useConfigState, useLoggedInState } from '../../context/AppContext';
import { useRouter } from 'next/router';
import { toIngestEndpointRevision } from '../../utils/NavigationPaths';
import { ChildrenAndIdProps } from '../../types/props/ChildrenAndIdProps';
import { analyticsEnabled } from '../../utils/AnalyticsUtils';
import { buildIngestEndpointsButtons } from './IngestEndpointSection';

const IngestEndpointRevisionSection: FC<ChildrenAndIdProps> = (props: ChildrenAndIdProps) => {
    const router = useRouter();
    const { id, children } = props;

    const { templateInteractions, orgUserState } = useLoggedInState();
    const { useSignup } = useConfigState();

    const { dispatchDialogAction } = templateInteractions;
    const currentOrgPermissions = extractPermissionsFromOrgUser(orgUserState);

    const sectionProps: SectionProps<NavIngestEndpointRevision> = {
        children,
        sectionKey: SectionKey.ingestEndpointRevision,
        queryResult: useQuery<NavIngestEndpointRevision>(NavIngestEndpointRevisionQuery, {
            variables: { id },
        }),
        initContext: (data) => {
            templateInteractions.setSectionHasAnalytics(
                analyticsEnabled(data.getIngestEndpointRevision.ingest_endpoint),
            );
        },
        buildButtonsProps: (data, orgPermissions) => [
            ...buildIngestEndpointsButtons(
                data.me.orgs,
                data.getIngestEndpointRevision.ingest_endpoint.data_manager_account.org,
                data.getIngestEndpointRevision.ingest_endpoint.data_manager_account,
                data.getIngestEndpointRevision.ingest_endpoint.data_manager_account.org
                    .tag_manager_account,
                data.getIngestEndpointRevision.ingest_endpoint.data_manager_account
                    .ingest_endpoints,
                data.getIngestEndpointRevision.ingest_endpoint,
                analyticsEnabled(data.getIngestEndpointRevision.ingest_endpoint),
                router,
                orgPermissions,
                useSignup,
                'Revisions',
            ),
            buildIngestEndpointRevisionButtonProps(
                router,
                data.getIngestEndpointRevision.ingest_endpoint.ingest_endpoint_revisions,
                id,
                data.getIngestEndpointRevision.name,
                true,
            ),
        ],
        buildMenuItemsProps: () => [],
        extractOrgUserDetails: (data) =>
            data.getIngestEndpointRevision.ingest_endpoint.data_manager_account.org,
        buildBreadcrumbActions: () => [
            buildDuplicateAction(
                () =>
                    pageActions.duplicateIngestEndpointRevision(
                        {
                            dispatchDialogAction,
                        },
                        id,
                        (
                            id: string,
                            pageRefresh: () => void,
                            handleDialogClose: (checkChanges: boolean) => void,
                        ) => {
                            handleDialogClose(false);
                            router.push(toIngestEndpointRevision({ id })).then();
                        },
                    ),
                `Duplicate Revision`,
                () => !currentOrgPermissions.canCreate,
            ),
        ],
        accountExpireIn: orgUserState?.dataManagerAccount?.trialExpiration ?? undefined,
        accountIsTrial: orgUserState?.dataManagerAccount?.isTrial ?? undefined,
    };

    return <Section<NavIngestEndpointRevision> {...sectionProps} />;
};

export default IngestEndpointRevisionSection;
