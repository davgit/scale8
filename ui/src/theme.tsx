import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
