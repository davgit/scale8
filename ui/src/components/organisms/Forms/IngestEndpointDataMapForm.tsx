import { ChangeEvent, FC, useEffect, useState } from 'react';
import { FormControlLabel, Radio, RadioGroup } from '@mui/material';
import ControlledTextInput from '../../atoms/ControlledInputs/ControlledTextInput';
import ControlledSelect from '../../atoms/ControlledInputs/ControlledSelect';
import {
    DataManagerDateTimeMacros,
    DataManagerIntegerMacros,
    DataManagerStringMacros,
    DataManagerTimestampMacros,
    VarType,
} from '../../../gql/generated/globalTypes';
import { DataMapDefaultValueFormSection } from '../../molecules/DataMapDefaultValueFormSection';
import { snakeToTitleCase } from '../../../utils/TextUtils';
import { SelectValueWithSub } from '../../../hooks/form/useFormValidation';
import { isVarTypeObject, isVarTypeScalar } from '../../../utils/VarTypeUtils';
import DrawerFormLayout from '../../molecules/DrawerFormLayout';
import ValidationRulesSection from '../../molecules/ValidationRulesSection';
import { IngestEndpointDataMapFormProps } from '../../../dialogPages/dataManager/IngestEndpointDataMapCreate';

const IngestEndpointDataMapForm: FC<IngestEndpointDataMapFormProps> = (
    props: IngestEndpointDataMapFormProps,
) => {
    const [useMacro, setUseMacro] = useState(false);

    const getRadioValue = () => {
        if (props.values.useDefault) {
            if (useMacro) {
                return 'macro';
            }
            return 'default';
        }
        if (props.values.optional) {
            return 'optional';
        }
        return 'required';
    };

    const getMacroValues = (): string[] => {
        if (props.values.varType === VarType.STRING) {
            return Object.values(DataManagerStringMacros);
        } else if (props.values.varType === VarType.INT) {
            return Object.values(DataManagerIntegerMacros);
        } else if (props.values.varType === VarType.FLOAT) {
            return [];
            // return Object.values(DataManagerFloatMacros);
        } else if (props.values.varType === VarType.BOOLEAN) {
            return [];
            // return Object.values(DataManagerBooleanMacros);
        } else if (props.values.varType === VarType.TIMESTAMP) {
            return Object.values(DataManagerTimestampMacros);
        } else if (props.values.varType === VarType.DATETIME) {
            return Object.values(DataManagerDateTimeMacros);
        } else {
            return [];
        }
    };

    const getMacroSelectValues = (): SelectValueWithSub[] => {
        return getMacroValues().map((_) => ({
            key: _,
            text: _,
        }));
    };

    const handleRadioChange = (event: ChangeEvent<HTMLInputElement>) => {
        const newValue = (event.target as HTMLInputElement).value;
        const useDefault = newValue === 'macro' || newValue === 'default';
        props.setValues({
            ...props.values,
            useDefault,
            optional: newValue === 'optional',
            defaultValue: props.values.varType === VarType.BOOLEAN ? false : '', // boolean select will default to '' otherwise
            defaultValues: [],
        });

        setUseMacro(newValue === 'macro');
    };

    const defaultValue = props.values.defaultValue;

    useEffect(() => {
        setUseMacro(getMacroValues().includes(defaultValue.toString()) ? true : useMacro);
    }, [defaultValue]);

    const varTypeRawValues = Object.keys(VarType)
        .filter((_) => _ !== 'NULL')
        .map((key) => ({
            key,
            text: VarType[key as unknown as keyof typeof VarType],
        }));

    const varTypeValues = varTypeRawValues.map((_) => ({
        key: _.key,
        text: snakeToTitleCase(_.text),
    }));

    const varTypeValuesLastLevel = varTypeRawValues
        .filter((_) => _.key !== VarType.OBJECT && _.key !== VarType.ARRAY_OBJECT)
        .map((_) => ({
            key: _.key,
            text: snakeToTitleCase(_.text),
        }));

    return (
        <DrawerFormLayout {...props} noSubmit={props.readOnly}>
            <ControlledTextInput
                name="key"
                label="Key"
                formProps={props}
                className="DrawerFormField"
                disabled={props.edit || props.readOnly}
                required
            />
            <ControlledSelect
                className="DrawerFormField"
                label="Var Type"
                name="varType"
                values={props.lastLevel ? varTypeValuesLastLevel : varTypeValues}
                formProps={props}
                disabled={props.edit || props.readOnly}
                required
            />
            <RadioGroup
                value={getRadioValue()}
                onChange={handleRadioChange}
                sx={{
                    width: '100%',
                    marginBottom: (theme) => theme.spacing(3),
                }}
            >
                <FormControlLabel
                    value="optional"
                    control={<Radio />}
                    label="Optional"
                    disabled={props.readOnly}
                />
                <FormControlLabel
                    value="required"
                    control={<Radio />}
                    label="Required"
                    disabled={props.readOnly}
                />
                {!isVarTypeObject(props.values.varType as VarType) && (
                    <FormControlLabel
                        value="default"
                        control={<Radio />}
                        label="Use Default Value"
                        disabled={props.readOnly}
                    />
                )}
                {isVarTypeScalar(props.values.varType as VarType) &&
                    getMacroValues().length !== 0 && (
                        <FormControlLabel
                            value="macro"
                            control={<Radio />}
                            label="Use Macro"
                            disabled={props.readOnly}
                        />
                    )}
            </RadioGroup>

            {props.values.varType !== '' &&
                props.values.useDefault &&
                (useMacro ? (
                    <ControlledSelect
                        className="DrawerFormField"
                        label="Macro"
                        name="defaultValue"
                        values={getMacroSelectValues()}
                        formProps={props}
                        disabled={props.readOnly}
                        required
                    />
                ) : (
                    <DataMapDefaultValueFormSection
                        varType={props.values.varType}
                        formProps={props}
                        readOnly={props.readOnly}
                    />
                ))}

            <ValidationRulesSection
                varType={props.values.varType === '' ? null : (props.values.varType as VarType)}
                validationRules={props.values.validationRules}
                handleChange={props.handleChange}
                readOnly={props.readOnly}
            />
        </DrawerFormLayout>
    );
};

export default IngestEndpointDataMapForm;
