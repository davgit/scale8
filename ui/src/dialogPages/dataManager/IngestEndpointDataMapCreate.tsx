import { Dispatch, FC, SetStateAction } from 'react';
import { FormProps, useFormValidation } from '../../hooks/form/useFormValidation';
import IngestEndpointDataMapForm from '../../components/organisms/Forms/IngestEndpointDataMapForm';
import {
    AddChildrenIngestEndpointDataMapsInput,
    IngestEndpointAddIngestEndpointDataMapsInput,
    VarType,
} from '../../gql/generated/globalTypes';
import { useMutation } from '@apollo/client';
import CreateIngestEndpointDataMapChildQuery from '../../gql/mutations/CreateIngestEndpointDataMapChildQuery';
import CreateIngestEndpointDataMapQuery from '../../gql/mutations/CreateIngestEndpointDataMapQuery';
import keyValidator from '../../utils/validators/keyValidator';
import { isVarTypeArray } from '../../utils/VarTypeUtils';
import { DialogPageProps } from '../../types/DialogTypes';
import { usePageDialogControls } from '../../hooks/dialog/usePageDialogControls';
import { buildStandardFormInfo } from '../../utils/InfoLabelsUtils';
import { PlatformDataMapValidation } from '../../types/DataMapsTypes';
import { logError } from '../../utils/logUtils';

export type IngestEndpointDataMapValues = {
    id?: string;
    key: string;
    varType: string;
    defaultValue: S8DataMapValue;
    defaultValues: S8DataMapValue[];
    optional: boolean;
    useDefault: boolean;
    validationRules: PlatformDataMapValidation[] | undefined;
};

export type IngestEndpointDataMapFormProps = FormProps<IngestEndpointDataMapValues> & {
    lastLevel: boolean;
    edit: boolean;
    readOnly: boolean;
};

export const IngestEndpointDataMapCreateWithLevel: FC<DialogPageProps & { lastLevel: boolean }> = (
    props: DialogPageProps & { lastLevel: boolean },
) => {
    const { id: parentId, contextId: ingest_endpoint_revision_id, name: parentName } = props;

    const isRoot = parentId === undefined || parentId === '';

    const [
        addIngestEndpointDataMaps,
        { loading: rootLoading, data: rootData, error: rootGqlError },
    ] = useMutation(CreateIngestEndpointDataMapQuery);

    const [
        addChildrenIngestEndpointDataMaps,
        { loading: childLoading, data: childData, error: childGqlError },
    ] = useMutation(CreateIngestEndpointDataMapChildQuery);

    const loading = isRoot ? rootLoading : childLoading;
    const data = isRoot ? rootData : childData;
    const gqlError = isRoot ? rootGqlError : childGqlError;

    const formInitialState: IngestEndpointDataMapValues = {
        key: '',
        varType: '',
        defaultValue: '',
        defaultValues: [],
        optional: true,
        useDefault: false,
        validationRules: undefined,
    };

    const submitForm = async (ingestEndpointDataMapValues: IngestEndpointDataMapValues) => {
        const varType = ingestEndpointDataMapValues.varType as VarType;
        const useArrayDefault = isVarTypeArray(varType);
        const ingest_endpoint_data_maps = [
            {
                key: ingestEndpointDataMapValues.key,
                var_type: varType,
                default_value:
                    ingestEndpointDataMapValues.useDefault && !useArrayDefault
                        ? ingestEndpointDataMapValues.defaultValue
                        : undefined,
                default_values:
                    ingestEndpointDataMapValues.useDefault && useArrayDefault
                        ? ingestEndpointDataMapValues.defaultValues
                        : undefined,
                optional: ingestEndpointDataMapValues.useDefault
                    ? false
                    : ingestEndpointDataMapValues.optional,
                validation_rules:
                    ingestEndpointDataMapValues.validationRules === undefined
                        ? undefined
                        : ingestEndpointDataMapValues.validationRules,
            },
        ];

        if (isRoot) {
            const ingestEndpointAddIngestEndpointDataMapsInput: IngestEndpointAddIngestEndpointDataMapsInput =
                {
                    ingest_endpoint_revision_id,
                    ingest_endpoint_data_maps,
                };
            try {
                await addIngestEndpointDataMaps({
                    variables: { ingestEndpointAddIngestEndpointDataMapsInput },
                });
            } catch (error) {
                logError(error);
            }
        } else {
            const addChildrenIngestEndpointDataMapsInput: AddChildrenIngestEndpointDataMapsInput = {
                ingest_endpoint_data_map_id: parentId,
                ingest_endpoint_data_maps,
            };
            try {
                await addChildrenIngestEndpointDataMaps({
                    variables: { addChildrenIngestEndpointDataMapsInput },
                });
            } catch (error) {
                logError(error);
            }
        }
    };

    const customValueSetter = (
        valueKey: string,
        value: any,
        values: IngestEndpointDataMapValues,
        setValues: Dispatch<SetStateAction<IngestEndpointDataMapValues>>,
    ) => {
        if (valueKey === 'varType') {
            setValues({
                ...values,
                [valueKey]: value,
                optional: true,
                useDefault: false,
                defaultValue: value === VarType.BOOLEAN ? false : '', // boolean select will default to '' otherwise
                defaultValues: [],
                validationRules: undefined,
            });
        } else {
            setValues({
                ...values,
                [valueKey]: value,
            });
        }
    };

    const formValidationValues = useFormValidation<IngestEndpointDataMapValues>(
        formInitialState,
        [
            {
                field: 'key',
                validator: keyValidator,
                error: () => 'Invalid key',
            },
        ],
        submitForm,
        false,
        customValueSetter,
    );

    const successfullySubmitted = isRoot
        ? data?.addIngestEndpointDataMaps[0].id !== undefined
        : data?.addChildrenIngestEndpointDataMaps[0].id !== undefined;

    usePageDialogControls(
        JSON.stringify(formInitialState) === JSON.stringify(formValidationValues.values),
        successfullySubmitted,
        props.setPageHasChanges,
        props.handleDialogClose,
        props.pageRefresh,
    );

    if (loading || successfullySubmitted) {
        return <div />;
    }

    const formProps: IngestEndpointDataMapFormProps = {
        ...formValidationValues,
        gqlError,
        submitText: 'Create Data Map',
        title: parentName ? `Add child data map to ${parentName}` : 'Create Data Map',
        lastLevel: props.lastLevel,
        formInfoProps: buildStandardFormInfo('ingestEndpointDataMaps', 'Add'),
        handleDialogClose: props.handleDialogClose,
        edit: false,
        readOnly: false,
    };

    return <IngestEndpointDataMapForm {...formProps} />;
};

const IngestEndpointDataMapCreate: FC<DialogPageProps> = (props: DialogPageProps) => {
    return <IngestEndpointDataMapCreateWithLevel lastLevel={false} {...props} />;
};

export default IngestEndpointDataMapCreate;
