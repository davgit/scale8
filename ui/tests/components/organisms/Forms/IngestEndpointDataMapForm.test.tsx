import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { FC } from 'react';
import { useMockFormProps } from '../../../componentsMocks/useMockFormProps';
import IngestEndpointDataMapForm from '../../../../src/components/organisms/Forms/IngestEndpointDataMapForm';
import { IngestEndpointDataMapValues } from '../../../../src/dialogPages/dataManager/IngestEndpointDataMapCreate';

const MockForm: FC = () => {
    const formProps = useMockFormProps<IngestEndpointDataMapValues>({
        key: '',
        varType: '',
        defaultValue: '',
        defaultValues: [],
        optional: false,
        useDefault: false,
        validationRules: undefined,
    });

    return (
        <IngestEndpointDataMapForm {...formProps} lastLevel={false} edit={false} readOnly={false} />
    );
};

describe('IngestEndpointDataMapForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
