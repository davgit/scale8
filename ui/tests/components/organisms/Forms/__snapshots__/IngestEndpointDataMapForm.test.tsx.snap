// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngestEndpointDataMapForm form render unchanged 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-s92abg"
  >
    <div
      class="MuiBox-root css-70qvj9"
    >
      <div
        class="MuiBox-root css-1isemmb"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-uix0gv"
      >
        mock form
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiBox-root css-yn9cjk"
    >
      <div
        class="MuiBox-root css-1676va5"
      >
        <form
          class="MuiBox-root css-mjj15b"
        >
          <div
            class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
            >
              Key
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1ejkwvn-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                name="key"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root DialogFormField css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
            >
              Var Type
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-1ejkwvn-MuiInputBase-root-MuiInput-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-varType"
                class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                id="mui-component-select-varType"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="varType"
                required=""
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-yqkyv6-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiFormGroup-root css-1datqi3-MuiFormGroup-root"
            role="radiogroup"
          >
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-nxeipd-MuiButtonBase-root-MuiRadio-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  name="optional-radio"
                  type="radio"
                  value="optional"
                />
                <span
                  class="css-hyxlzm"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17s4f6u-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ba0akt-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
              >
                Optional
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root Mui-checked css-nxeipd-MuiButtonBase-root-MuiRadio-root"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  name="required-radio"
                  type="radio"
                  value="required"
                />
                <span
                  class="css-hyxlzm"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17s4f6u-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-40zptw-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
              >
                Required
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-nxeipd-MuiButtonBase-root-MuiRadio-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  name="default-radio"
                  type="radio"
                  value="default"
                />
                <span
                  class="css-hyxlzm"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17s4f6u-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ba0akt-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
              >
                Use Default Value
              </span>
            </label>
          </div>
          <div
            class="MuiBox-root css-wpygli"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-xcbv7o-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
